3-enrich/stream-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/kinesis/sources/AbstractSource.scala

```diff
@@ -191,7 +191,16 @@ abstract class AbstractSource(config: KinesisEnrichConfig, igluResolver: Resolve
       validatedMaybeEvent match {
         case Success(co) => (tabSeparateEnrichedEvent(co), if (config.useIpAddressAsPartitionKey) {
             co.user_ipaddress
-          } else {
+          } 
+	  else if (config.partitionKey != null) {
+          try
+            String.valueOf(co.getClass.getDeclaredField(config.partitionKey).get(co))
+          catch {
+            case _: Exception =>
```

Comment:
Would also like to see some tests if this is going to be extracted into separate function.